### PR TITLE
Allow json editor in grid to expand to json side panel editor

### PR DESCRIPTION
**Pull request by joshenlim**
image

### CODE REVIEW

File: studio/components/grid/types/grid.ts [hunk 1]
```diff
@@ -12,63 +12,34 @@ export interface GridProps {
 }
 
 export interface SupabaseGridProps {
-  /**
-   * database table swagger or table name
-   */
+  // database table swagger or table name
```
Comment (alaister) [inline studio/components/grid/types/grid.ts:15]: Any reason for changing these comment types? I think the editor can read `/**` comments and show tooltips when we're developing

File: studio/components/interfaces/TableGridEditor/SidePanelEditor/RowEditor/JsonEditor/JsonEditor.tsx [hunk 2]
```diff
@@ -72,7 +84,15 @@ const JsonEdit: FC<JsonEditProps> = ({ column, jsonString, visible, closePanel,
       }
       visible={visible}
       onCancel={closePanel}
-      customFooter={<ActionBar closePanel={closePanel} applyFunction={validateJSON} />}
+      customFooter={
+        <ActionBar
+          // loading
```
Comment (alaister) [inline studio/components/interfaces/TableGridEditor/SidePanelEditor/RowEditor/JsonEditor/JsonEditor.tsx:89]: loading?

File: studio/components/grid/components/common/BlockKeys.tsx [hunk 3]
```diff
@@ -33,7 +33,10 @@ export const BlockKeys: FC<BlockKeysProps> = ({ value, children, onEscape, onEnt
   )
 
   function onBlur() {
-    if (onEnter) onEnter(value)
+    // [Joshen] Commenting this out for now as its causing some odd behavior
+    // where its triggering when a children of the Popover overlay component is clicked
+    // Also to make the save action a bit more deliberate
+    // if (onEnter) onEnter(value)
```
Comment (alaister) [inline studio/components/grid/components/common/BlockKeys.tsx:39]: Probably best to do in another PR, but I think it would be nice to prevent the json editor from closing if there are unsaved changes and the user blurs it. I found myself making a few changes, and then accidentally clicking outside of the editor and losing them!

**Review by alaister — COMMENTED** [review]
Very nice indeed!
Just a few comments:

File: studio/components/grid/components/common/BlockKeys.tsx (hunk 3 above)
Comment (joshenlim) [reply studio/components/grid/components/common/BlockKeys.tsx:39]: gotcha will take a look into this one - its tricky cause the onBlur is getting triggered whenever BlockKey has a Button component rendered in it or when i click anywhere in this space
image

hence why i commented it out for now

File: studio/components/grid/types/grid.ts (hunk 1 above)
Comment (joshenlim) [reply studio/components/grid/types/grid.ts:15]: ahhh i see - no real reason tbh, i'll revert it

File: studio/components/grid/components/common/BlockKeys.tsx (hunk 3 above)
Comment (joshenlim) [reply studio/components/grid/components/common/BlockKeys.tsx:39]: hmm okay this turns out to be a little more tricky because we're using the popover component which automatically closes once we click outside of it.

for now - i've found a workaround for the above issue i raised and i'll reinstate the previous behaviour where losing focus of the editor will save the changes

**Review by alaister — APPROVED** [review]
LGTM!
